netshare/configs/__init__.py

```diff
@@ -0,0 +1,51 @@
+import os
+from typing import Any, Optional, Union
+
+from config_io import Config
+
+from netshare.utils.logger import logger
+
+from .default import __path__ as defaults_path
+
+_config: Optional[Config] = None
+
+
+def load_from_file(path: str) -> None:
+    global _config
+    _config = Config.load_from_file(path, default_search_paths=defaults_path)
+
+
+def set_config(config: Union[Config, dict]) -> None:
+    global _config
+    _config = config
+
+
+def get_config(
```

Comment:
I understand that `path2` is for providing an alternative path in the config for supporting the use case where a config may appear either in global_config or its own section. But this solution (adding a parameter `path2` and an ad-hoc `if path2:`) is not very elegant. Also, it's hard to use as now users need to specify two parameters "path" and "path2". In addition, if a parameter was originally in its own section, and somehow later on we need to share it with other sections and therefore move it to `global_config` section, we will need to modify all places where we get this config.

One alternative is that when we load the config, we populate the configs in `global` section into all other sections. We can then remove `path2`, and when we want to `get_config`, we always use the path in its own section. One downside of this approach is that we do not have any isolation for the configs in `global` section (i.e., global config may be populated into sections that do not need it). But given that we do not have isolation in the current version anyways (all functions can access any part of the configs, including the ones in `global`), I think it is fine for now.

Reply:
I see there is another use case of `path2`, which is for specifying multiple config path options (e.g., for backward compatibility). Even in that case, I think it is still not elegant to use `path2` to support that. What if we have a third possibility? Definitely, we do not want to add another `path3`. 

For this use case, we can just manually use multiple calls of `get_config`. If we really want to accomplish it in one call, we can make `path` either a string or a list of strings (each element in the list is one possibility of the path). I would prefer the first option to make the API cleaner.

Reply:
That's an interesting discussion!
I added `path2` only to allow alternative paths to find the relevant configuration (as you thought in the second comment).
I choose to use `path2` and not `List[str]` because I added it only for backward compatibility (i.e. that old configurations will still work on the current branch).
I agree that this is not fully clear. Here are my two suggestions:
1. We can change the naming here to make it more clear. WDYT about changing the name of `path2` to `backward_compatible_path`?
2. Alternatively, I prefer to change it to a general list of paths (instead of asking the user to call this function multiple times). I think that it will be a cleaner API / easier to use.
I'm really against moving configuration during runtime - it makes the flow much harder to follow. I.e. when a component wants to use the configuration, it doesn't know where to search it, because something has moved it around. It is much much easier to assume that the input config is *exactly* what you get.

Reply:
Got it. That makes sense to me now! Thanks for the explanations! Both approaches you suggested sound great to me! I would probably slightly prefer the second one, just in case in the future we update the config format again and the code will need to support more than 3 possibilities.

Reply:
ok, then the second one it is :)